api/app/controllers/spree/api/base_controller.rb

```diff
@@ -188,6 +189,11 @@ def paginate(resource)
       def default_per_page
         Kaminari.config.default_per_page
       end
+
+      def invalid_transition(error)
+        logger.error("invalid_transition #{error.event} from #{error.from} for #{error.object.class.name}. Error: #{error.inspect}")
+        respond_with(@order, default_template: 'spree/api/orders/could_not_transition', status: 422)
```

Comment:
I think `@order` is not present in any controller that could raise a state machine invalid transition. I like to be generic but can't work only with orders if we have other state machines in the system.

Reply:
Thank you for catching that! @kennyadsl 

I'll change `@order` with `error.object` which should do the trick.

Reply:
Yes, but we should also have a generic template, now it's `spree/api/orders/could_not_transition`.

Reply:
@kennyadsl changed and split into three different commits.

I moved `spree/api/orders/could_not_transition` under errors `spree/api/errors/could_not_transition`. The changes in the controller should not diff in the api response.